assets/wizards/advertising/components/onboarding/index.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * WordPress dependencies.
+ */
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { useEffect, useState, useRef, Fragment } from '@wordpress/element';
+
+/**
+ * Internal dependencies.
+ */
+import { Card, ButtonCard, Notice, TextControl } from '../../../../components/src';
+import GoogleOAuth from '../../../connections/views/main/google';
+
+export default function AdsOnboarding( { onUpdate, onSuccess } ) {
+	const credentialsInputFile = useRef( null );
+	const [ fileError, setFileError ] = useState( '' );
+	const [ inFlight, setInFlight ] = useState( false );
+	const [ networkCode, setNetworkCode ] = useState( '' );
+	const [ isConnected, setIsConnected ] = useState( false );
+	const [ useOAuth, setUseOAuth ] = useState( null );
+
+	const updateGAMCredentials = credentials => {
+		setInFlight( true );
+		apiFetch( {
+			path: '/newspack/v1/wizard/advertising/credentials',
+			method: 'post',
+			data: { credentials, onboarding: true },
+		} )
+			.then( () => {
+				setIsConnected( true );
+				if ( typeof onSuccess === 'function' ) onSuccess();
+			} )
+			.finally( () => {
+				setInFlight( false );
+			} );
+	};
+
+	const handleCredentialsFile = event => {
+		if ( event.target.files.length && event.target.files[ 0 ] ) {
+			const reader = new FileReader();
+			reader.readAsText( event.target.files[ 0 ], 'UTF-8' );
+			reader.onload = function ( ev ) {
+				let credentials;
+				try {
+					credentials = JSON.parse( ev.target.result );
+				} catch ( error ) {
+					setFileError( __( 'Invalid JSON file', 'newspack' ) );
+					return;
+				}
+				updateGAMCredentials( credentials );
```

Comment:
If credentials are supplied before network code is entered, there's no way to enter it – the modal is dismissed. The modal should disappear only after all conditions are satisfied.

Reply:
All conditions are optional. If a service account credential is uploaded, the network code is fetched from the credential. I'd rather not have any blocker to exit this modal, since any inserted value is enough to use GAM (legacy with network code or connected through either OAuth or Service Account).

Reply:
That's right, I forgot the credentials contain the network code.